src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -2,20 +2,96 @@
 
 import org.junit.Before;
 import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
 
-import static org.junit.Assert.assertTrue;
+import java.math.BigDecimal;
+import java.util.NoSuchElementException;
 
-public class ArrayListProductDaoTest
-{
-    private ProductDao productDao;
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+public class ArrayListProductDaoTest {
+    private ArrayListProductDao productDao;
+
+    @Mock
+    private Product product1;
+    @Mock
+    private Product product2;
+    @Mock
+    private Product product3;
 
     @Before
     public void setup() {
+        MockitoAnnotations.initMocks(this);
+
         productDao = new ArrayListProductDao();
+        productDao.getProductList().add(product1);
+
+        when(product1.getId()).thenReturn(1L);
+        when(product1.getPrice()).thenReturn(new BigDecimal(100));
+        when(product1.getStock()).thenReturn(100);
+
+        when(product3.getId()).thenReturn(3L);
+        when(product3.getPrice()).thenReturn(new BigDecimal(300));
+        when(product3.getStock()).thenReturn(300);
     }
 
     @Test
-    public void testFindProductsNoResults() {
-        assertTrue(productDao.findProducts().isEmpty());
+    public void testGetProduct() {
+        product2 = productDao.getProduct(1L);
+        assertEquals(product1.getId(), product2.getId());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testGetProductNullId() {
+        productDao.getProduct(null);
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    public void testGetNonExistentProduct() {
+        productDao.getProduct(2L);
+    }
+
+    @Test
+    public void testFindProducts() {
+        assertTrue(productDao.findProducts().size() == 1
+                && productDao.findProducts().contains(product1)
+                && productDao.findProducts().get(0).getPrice() != null
+                && productDao.findProducts().get(0).getStock() > 0);
+    }
+
+    @Test
+    public void testSave() {
+        productDao.save(product3);
+        assertTrue(productDao.findProducts().contains(product3));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testSaveAlreadyExistentProduct() {
+        productDao.save(product1);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testSaveProductNullId() {
+        productDao.save(null);
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    public void testDelete() {
+        product2 = productDao.getProduct(product1.getId());
+        productDao.delete(product2.getId());
+        productDao.getProduct(product1.getId());
```

Comment:
Rewrite this test, just delete by '1L' id and check that the list has become empty